src/include/grammar/grammar.h

```diff
@@ -0,0 +1,42 @@
+#ifndef JUCC_GRAMMAR_H
+#define JUCC_GRAMMAR_H
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace jucc {
+namespace grammar {
+const char EPSILON[] = "epsilon";
+typedef std::vector<std::string> Entity;
+
+class Rule {
+  std::vector<std::string> entities;
+
+ public:
+  Rule() = default;
+  explicit Rule(std::vector<std::string> entities) : entities(std::move(entities)) {}
+  [[nodiscard]] const std::vector<std::string> &getEntities() const { return entities; }
+  void setEntities(const std::vector<std::string> &entities) { Rule::entities = entities; }
+  [[nodiscard]] std::string toString() const;
+  [[nodiscard]] bool HasPrefix(const Entity &) const;
+};
+
+typedef std::vector<grammar::Rule> Rules;
+class Production {
+  std::string parent;
+  std::vector<Rule> rules;
+
+ public:
+  Production() = default;
+  Production(std::string parent, std::vector<Rule> rules) : parent(std::move(parent)), rules(std::move(rules)) {}
+  [[nodiscard]] const std::string &getParent() const { return parent; }
+  [[nodiscard]] const std::vector<Rule> &getRules() const { return rules; }
+  void setParent(const std::string &parent) { Production::parent = parent; }
+  void setRules(const std::vector<Rule> &rules) { Production::rules = rules; }
+};
+
```

Comment:
Please write more comments.
Please make the codebase user-friendly.
Thanks.

Reply:
I have added comments on main rebase will fix this.